server/neptune/temporalworker/server.go

```diff
@@ -0,0 +1,192 @@
+	SslKeyFile       string
+	TemporalHostPort string
+	Scope            tally.Scope
+	Closer           io.Closer
+}
+
+type httpServerProxy struct {
+	*http.Server
+	SSLCertFile string
+	SSLKeyFile  string
+	Logger      logging.Logger
+}
+
+func (p *httpServerProxy) ListenAndServe() {
+	var err error
+	if p.SSLCertFile != "" && p.SSLKeyFile != "" {
+		err = p.Server.ListenAndServeTLS(p.SSLCertFile, p.SSLKeyFile)
+	} else {
+		err = p.Server.ListenAndServe()
+	}
+	if err != nil && err != http.ErrServerClosed {
+		p.Logger.Error(err.Error())
+	}
+}
+
+type Server struct {
+	Logger           logging.Logger
+	SSLCertFile      string
+	SSLKeyFile       string
+	Negroni          *negroni.Negroni
+	LyftMode         server.Mode
+	Port             int
+	StatsScope       tally.Scope
+	StatsCloser      io.Closer
+	TemporalHostPort string
+	JobsController   *controllers.JobsController
+}
+
+// TODO: as more behavior is added into the TemporalWorker package, inject corresponding dependencies
+func NewServer(config *Config) (*Server, error) {
+	jobsController := &controllers.JobsController{
+		StatsScope: config.Scope,
+	}
+	// router initialization
+	router := mux.NewRouter()
+	router.HandleFunc("/healthz", Healthz).Methods("GET")
+	router.HandleFunc("/jobs/{job-id}", jobsController.GetProjectJobs).Methods("GET").Name(ProjectJobsViewRouteName)
+	router.HandleFunc("/jobs/{job-id}/ws", jobsController.GetProjectJobsWS).Methods("GET")
+	n := negroni.New(&negroni.Recovery{
+		Logger:     log.New(os.Stdout, "", log.LstdFlags),
+		PrintStack: false,
+		StackAll:   false,
+		StackSize:  1024 * 8,
+	})
+	n.UseHandler(router)
+	server := Server{
+		TemporalHostPort: config.TemporalHostPort,
+		Logger:           config.CtxLogger,
+		SSLCertFile:      config.SslCertFile,
+		SSLKeyFile:       config.SslKeyFile,
+		StatsScope:       config.Scope,
+		StatsCloser:      config.Closer,
+		Negroni:          n,
+		JobsController:   jobsController,
+	}
+	return &server, nil
+}
+
+func (s Server) Start() error {
+	defer s.Logger.Close()
+
+	// temporal client + worker initialization
+	temporalClient, err := s.buildTemporalClient()
+	if err != nil {
+		return err
+	}
+	defer temporalClient.Close()
+	go func() {
+		w := worker.New(temporalClient, DeployTaskqueue, worker.Options{
+			// ensures that sessions are preserved on the same worker
+			EnableSessionWorker: true,
+		})
+		if err := w.Run(worker.InterruptCh()); err != nil {
+			log.Fatalln("unable to start worker", err)
+		}
+	}()
+
+	// Ensure server gracefully drains connections when stopped.
+	stop := make(chan os.Signal, 1)
+	// Stop on SIGINTs and SIGTERMs.
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+
+	s.Logger.Info(fmt.Sprintf("Atlantis started - listening on port %v", s.Port))
+	httpServerProxy := httpServerProxy{
+		SSLCertFile: s.SSLCertFile,
+		SSLKeyFile:  s.SSLKeyFile,
+		Server:      &http.Server{Addr: fmt.Sprintf(":%d", s.Port), Handler: s.Negroni},
+		Logger:      s.Logger,
+	}
```

Comment:
you can initialize this in NewServer as well right? and this can just be called server with interface type http.Server